src/Block/Types.hs

```diff
@@ -14,24 +14,44 @@ module Block.Types (
   OgmiosResponse (..),
   TxOut (..),
   BlockInfo (..),
+  blockId,
+  blockSlot,
 ) where
 
 import Data.Aeson (FromJSON, ToJSON, withObject, (.:), (.:?))
 import Data.Aeson qualified as Aeson
+import Data.ByteString.Lazy (toStrict)
 import Data.HashMap.Strict qualified as HashMap
 import Data.Int (Int64)
 import Data.Map (Map)
 import Data.Map qualified as Map
 import Data.Text (Text)
+import Data.Text.Encoding (decodeUtf8)
 import GHC.Exts (toList)
 import GHC.Generics (Generic)
 
-data BlockInfo = BlockInfo
-  { blockSlot :: Int64
-  , blockId :: Text
-  }
+data BlockInfo = BlockInfo Int64 Text | BlockOrigin
   deriving stock (Generic, Show)
-  deriving anyclass (ToJSON)
+
+instance ToJSON BlockInfo where
+  toJSON BlockOrigin =
+    Aeson.object
+      [ "blockSlot" Aeson..= (0 :: Int64)
+      , "blockId" Aeson..= ("0" :: Text)
+      ]
+  toJSON (BlockInfo slot' id') =
+    Aeson.object
+      [ "blockSlot" Aeson..= slot'
+      , "blockId" Aeson..= id'
+      ]
+
+blockSlot :: BlockInfo -> Int64
```

Comment:
After changes in `Database` module, this wouldn't be necessary to keep

Reply:
Don't think I can entirely get rid of this since its necessary for the Config.hs

Reply:
I genuinely start to hate how current config file works, we should consider entirely moving to CLI parameters. Lack of easily defined mutually exclusive options starts to cause problems